### DISCUSSION TITLE
From 7.4 to 7.5 *** KeyError Exception: key not found: :id_sort

### DISCUSSION THREAD

**Discussion by robinmonjo** [mongodb/mongoid#5520] [Q&A]
Hello all,

I'm upgrading mongoid and I face a strange issue. All call to `criteria.find` that uses to work fine on 7.4 stopped working with this error:

```
*** KeyError Exception: key not found: :id_sort
```

Tracking down the issue it happens here https://github.com/mongodb/mongoid/blob/7.5-stable/lib/mongoid/contextual/mongo.rb#L272 in the find function.

Here we have:

```
          if sort = view.sort || ({ _id: 1 } unless limit_or_opts.try(:fetch, :id_sort) == :none)
            sorted_view = view.sort(sort)
          end
```

For me, limit_or_opts is `{}`. Performing `limit_or_opts.try(:fetch, :id_sort)` trigger the error (ruby 2.7 and 3).

Changing this code to:

`limit_or_opts.try(:[], :id_sort) == :none` would fix this error.

What can I do ?

## Edit 

Seems that for some reasons my call to find inject an empty hash as parameters (I'll have to track that down I don't know where it's coming from). But the issue can be reproduced with this small example:

```
require 'mongoid'

Mongoid.configure do |config|
  config.clients.default = {
    hosts: ['localhost:27017'],
    database: 'test-bug-db'
  }

  config.log_level = :debug
end

class Test
  include Mongoid::Document
  include Mongoid::Timestamps

  field :name
end


t = Test.create(name: "hello")
puts Test.where(name: "hello").first({}) # the {} parameter make the code crash on 7.5, not on 7.4

```

**Comment by alexbevi** [comment]
Hi @robinmonjo, it seems `:id_sort` was removed then re-added in #5358. To help me investigate the behavior you've described can you share a self-contained reproduction that works in Mongoid 7.4 but now fails in 7.5?

**Reply by robinmonjo** [reply]
Thank you @alexbevi just added more information and a code sample

**Reply by alexbevi** [reply]
Much appreciated @robinmonjo. Once I've validated this I'll open a ticket for the team to follow up on and share the link here as well.

**Comment by alexbevi** [comment]
@robinmonjo I've opened [MONGOID-5546](https://jira.mongodb.org/browse/MONGOID-5546) for our team to investigate this further.

**Reply by robinmonjo** [reply]
Awesome, thank you 😊
